Gemfile

```diff
@@ -147,6 +147,7 @@ def override_gem(name, *args)
     dependencies.delete(dependency)
 
     gem name, *args
+    Bundler::UI::Shell.new.warn "Caution, using override_gem for #{name} with args: #{args.inspect}, review Gemfile.dev.rb if this is not desired!"
```

Comment:
* Ok, I'll move this up.
* Ok, I'll put an environment check
* I tried the caller route to find the file calling it but it's eval'd in bundler's dsl so the callstack is icky and may change.  I can perhaps filter out any callers that are from bundler land.